### PR TITLE
Minor cleanup in AsyncResponsePacketHandler.Thread.doRun

### PR DESCRIPTION
**Pull request by pveentjer**

### CODE REVIEW

**Comment by pveentjer** [comment]
run-lab-run

**Comment by pveentjer** [comment]
run-lab-run

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/AsyncResponsePacketHandler.java [hunk 1]
```diff
@@ -97,40 +98,26 @@ public ResponseThread(HazelcastThreadGroup threadGroup,
         public void run() {
             try {
                 doRun();
+            } catch (InterruptedException e) {
+                ignore(e);
```
Comment (mtopolnik) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/AsyncResponsePacketHandler.java]: Maybe an explicit `return` is a better idea? The exception isn't literally "ignored", it's being _handled_ by ending the runnable.

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/AsyncResponsePacketHandler.java [hunk 2]
```diff
@@ -97,40 +98,26 @@ public ResponseThread(HazelcastThreadGroup threadGroup,
...
-                Packet responsePacket;
-                try {
-                    responsePacket = workQueue.take();
-                } catch (InterruptedException e) {
-                    if (shutdown) {
-                        return;
-                    }
-                    continue;
-                }
+        @SuppressFBWarnings("VO_VOLATILE_INCREMENT")
```
Comment (mtopolnik) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/AsyncResponsePacketHandler.java]: An explicit `justification=` is a good idea here (even if seemingly obvious).

**Comment by mtopolnik** [comment]
:+1:

File: hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/AsyncResponsePacketHandler.java (hunk 2 above)
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/spi/impl/operationservice/impl/AsyncResponsePacketHandler.java]: The counter can be removed completely

**Pull request merged**